### PR TITLE
Remove internal EventPipe API and redundant tests 

### PR DESCRIPTION
**Pull request by sywhang**
This removes the internal `EventPipe` type which was being used to turn on EventPipe session by an internal profiler team via reflection. They no longer depend on this as they've moved onto using the diagnostics client library.

Also this removes a few redundant tests that are using this internal API via reflection to enable tracing tests. I went and checked all of these tests and verified that they are either:
- Testing the same scenario as the tests in tracing/eventpipe are testing
- Testing the same scenario as the BCL EventSource tests are testing
- Testing the internal reflection scenarios

The only test that is not already being tested by another tracing test is JittingStarted test which checks for whether a JittingStarted event is correctly received. I will file a separate issue to track adding this test under tracing/eventpipe.

cc @lakshanf After this PR, you should be able to safely remove the System.Diagnostics.Tracing.EventPipe type from rooted types for the ILTrim.

### CODE REVIEW

**Comment by josalem** [comment]
CC @lateralusX

Glad to see these are no longer needed!

**Review by LakshanF — APPROVED** [review]
Trimming HelloWorld app works with this change.

File: src/tests/tracing/tracevalidation/inducedgc/InducedGC.cs [hunk 1]
```diff
@@ -1,70 +0,0 @@
...
-using Tracing.Tests.Common;
-using Microsoft.Diagnostics.Tracing;
-using Microsoft.Diagnostics.Tracing.Parsers;
-using Microsoft.Diagnostics.Tracing.Parsers.Clr;
-
-namespace Tracing.Tests
-{
-    public static class TraceValidationInducedGC
-    {
-        private static int InducedGCIterations = 10;
```
Comment (josalem) [inline src/tests/tracing/tracevalidation/inducedgc/InducedGC.cs:15]: I'm not sure we have an equivalent test for the moder infrastructure.  We may want to add one, but that's not part of this PR.

**Review by josalem — APPROVED** [review]
The libraries test failure doesn't seem related to this patch.  LGTM 🚀
